doc/design.md

```diff
@@ -185,4 +185,116 @@ We hope Fluid users could represent it by the following Python/Fluid code.
 
 ```python
 def build_docker_image_from_git_source(
+        docker_source: "input,git",
+        built_image: "output,image",
+        path_to_dockerfile="/workspace/docker-source/Dockerfile",
+        path_to_context="/workspace/docker-source"):
+    '''Define a Tekton Task that builds a Docker image from a Git repo'''
+    couler.step(image="gcr.io/kaniko-project/executor:v0.14.0",
+               cmd=["/kaniko/executor"],
+               args=[f"--dockerfile={path_to_dockerfile}",
+                     f"--destination={built_image.url}",
+                     f"--context={path_to_context}"],
+               env={"DOCKER_CONFIG": "/tekton/home/.docker/"})
+```
+
+### Pipeline
+
+A Pipeline object is like function declaration.
+
+A Pipeline in Tekton defined an ordered series of Tasks. Users can specify whether
+the output of a `Task` is used as an input for the next `Task` using `from` property on `PipelineResources`
+
+As the following example comes from [Tekton's tutorial](https://github.com/tektoncd/pipeline/blob/master/docs/tutorial.md#creating-and-running-a-pipeline)
+
+``` yaml
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
```

Comment:
What is the lifecycle of a pipeline object?

Reply:
`Pipeline` object is like function definition in Python which includes `Pipeline Tasks`
A `PipelineRun` object would invoke the `Pipeline Tasks` as the dependency, can find some information from [Task Status](https://github.com/tektoncd/pipeline/blob/master/docs/taskruns.md#status)